### PR TITLE
Asking the user to select a storage option dialog for first time

### LINKED ISSUE #2068: At first download, ask to choose beetween "internal" and "external"
This is an idea to avoid users to download to the "wrong" place because for the beginners it is not obvious what are all the options the app provides.

### PR DESCRIPTION
**Pull request by markZuckerberg2**
Fixes #2068 
 **Description**
  Added a dialog which checks if the  user has selected internal or external storage. If they are not it will prompt a dialog to      select storage options. If they are selected already then it will start downloading instead of showing dialog.
**ScreenShots**
NA
**Testing**
It didn't break anything.

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt [hunk 1]
```diff
@@ -129,6 +129,11 @@ class SharedPreferenceUtil @Inject constructor(val context: Context) {
     set(prefShowBookmarksFromCurrentBook) = sharedPreferences.edit {
       putBoolean(PREF_SHOW_BOOKMARKS_ALL_BOOKS, prefShowBookmarksFromCurrentBook)
     }
+  var showStorageOption: Boolean
+    get() = sharedPreferences.getBoolean(PREF_SHOW_STORAGE_OPTION, false)
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt:133]: I think that default should be true. And after users select the default storage option then it has to store false.

**Review by gouri-panda — CHANGES_REQUESTED** [review]

**Comment by gouri-panda** [comment]
@kelson42 What do you think where we should show a dialog about internal and external storage option
1. Show when the user first goes to the download section like the cellular network warning.
2. When the User selected a zim file to download.

File: app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt [hunk 2]
```diff
@@ -247,17 +247,21 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
...
+        } else {
+          availableSpaceCalculator.hasAvailableSpaceFor(item,
+            { downloadFile(item.book) },
+            {
+              libraryList.snack(
+                getString(R.string.download_no_space) +
+                  "\n" + getString(R.string.space_available) + " " +
+                  it,
+                R.string.download_change_storage,
+                ::showStorageSelectDialog
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt:261]: The code is not formatted use `option` + `enter` in mac or `alt` + `enter` in windows or Linux to fix this.

**Review by gouri-panda — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt (hunk 2 above)
Comment (s-ayush2903) [reply app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt:261]: Nope, you're incorrect, the default binding is: `Ctrl/Cmd + Alt + L`

Comment (gouri-panda) [reply app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt:261]: @s-ayush2903 What I am saying is let the android studio fix the format issue.

Comment (s-ayush2903) [reply app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt:261]: I also told the same : ) 
What you told, doesn't fix formatting, instead it does some minor fix in the code(which isn't related to formatting)

File: app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import javax.inject.Inject
+
+/**
+ * WebServerHelper class is used to set up the suitable environment i.e. getting the
+ * ip address and port no. before starting the WebServer
+ * Created by Adeel Zafar on 18/07/2019.
+ */
+class WebServerHelper @Inject constructor(
+  private val kiwixServerFactory: KiwixServer.Factory,
+  private val ipAddressCallbacks: IpAddressCallbacks
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.kt:40]: This file is irrelevant to this PR. Please remove this.

**Review by gouri-panda — CHANGES_REQUESTED** [review]

**Comment by kelson42** [comment]
> @kelson42 What do you think where we should show a dialog about internal and external storage option
> 
>     1. Show when the user first goes to the download section like the cellular network warning.
> 
>     2. When the User selected a zim file to download.

I believe a dialog should be displayed if:
* the storage has not been configured yet AND
* Many storages with enough free space are available

It should happen once at the first download. Then the choice will be remembered.

Message should looks like "Download book to internal storage?" With two buttons "Configure" and "Yes/OK".

**Comment by gouri-panda** [comment]
>Many storages with enough free space are available
>Message should looks like "Download book to internal storage?" With two buttons "Configure" and "Yes/OK".
I just have one more doubts. Should we ask the download storage options that have enough space?

**Comment by kelson42** [comment]
> Should we ask the download storage options that have enough space?

Ask whom? When? How? If all of this happens automatically, then the answer is "yes".

**Comment by gouri-panda** [comment]
@markZuckerberg2 You have to modify this PR  a little bit. See the above comment [comment](https://github.com/kiwix/kiwix-android/pull/2624#issuecomment-813025562). If you have any further doubts or queries ask here or our slack channel. Thanks!

**Pull request closed without merging**